inc/c_util/two_d_array_ll.h

```diff
@@ -7,18 +7,248 @@
+            if (two_d_array->row_arrays[row_index] != NULL)                                                                                                                                  \
+            {                                                                                                                                                                                \
+                /* Codes_SRS_TWO_D_ARRAY_07_016: [ If the row specified by row_index has already been allocated, TWO_D_ARRAY_ALLOCATE_NEW_ROW(T) shall fail and return a non-zero value. ]*/ \
+                result = MU_FAILURE;                                                                                                                                                         \
+                LogError("Invalid arguments: uint64_t row_index=%" PRIu64 " has already been allocated", row_index);                                                                         \
+            }                                                                                                                                                                                \
+            else                                                                                                                                                                             \
+            {                                                                                                                                                                                \
+                /* Codes_SRS_TWO_D_ARRAY_07_017: [ Otherwise, TWO_D_ARRAY_ALLOCATE_NEW_ROW(T) shall allocate memory for the new row and return zero on success. ]*/                          \
+                T* new_row = malloc_2(two_d_array->cols, sizeof(T));                                                                                                                         \
+                if (new_row == NULL)                                                                                                                                                         \
+                {                                                                                                                                                                            \
+                    /* Codes_SRS_TWO_D_ARRAY_07_018: [ If there are any errors then TWO_D_ARRAY_ALLOCATE_NEW_ROW(T) shall fail and return a non-zero value. ]*/                              \
+                    LogError("failure in malloc_2(number of cols=%" PRIu64 ", sizeof(" MU_TOSTRING(T) ")=%zu)",                                                                              \
+                    two_d_array->cols, sizeof(T));                                                                                                                                           \
```

Comment:
>two_d_array->cols, sizeof(T));     [](http://example.com/codeflow?start=20&length=34)

This should be indented, since it belongs to the line above.

Reply:
done